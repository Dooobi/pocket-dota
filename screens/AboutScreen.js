import React from 'react';
import { View, Text } from 'react-native'

import ButtonHamburger from '../components/ButtonHamburger'
import { SCREEN_LABELS } from '../constants/Constants';

export default class Screen extends React.Component {
  static navigationOptions = ({ navigation }) => ({
    title: SCREEN_LABELS.ABOUT,
    headerRight: <ButtonHamburger />,
  });

  render() {
    return (
      <View>
        <Text>About and Legal content</Text>
      </View>
    );
  }
}

