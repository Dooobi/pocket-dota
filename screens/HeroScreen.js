import React from 'react';
import { View, Image, StyleSheet } from 'react-native';


import ButtonHamburger from '../components/ButtonHamburger';
import { Text, Container } from '../components/ui';
import Abilities from '../components/Hero/Abilities';
import Layout from '../constants/Layout';
import Colors from '../constants/Colors';
import { model_hero, model_hero_attributes } from '../constants/Models';
import Attributes from '../components/Hero/Attributes';
import ImageVert from '../components/Hero/ImageVert';
import Card from '../components/ui/Card';


export default class HeroScreen extends React.Component {
  static navigationOptions = ({ navigation }) => ({
    title: navigation.getParam('hero').name,
    headerRight: <ButtonHamburger />,
  });

  render() {
    const { name, bio, hype, tag, abilities } =  this.props.navigation.getParam('hero');
    const hero = model_hero(this.props.navigation.getParam('hero'));
    const attributes = model_hero_attributes(hero.attributes);


    return (
      <Container scrollable padInner style={styles.container} >

        <Card style={styles.row} collapsedTitle='Hype and Stats'>
          <Text>{hype}</Text>

          <View style={styles.imgHeroWrapper}>
            <ImageVert />
            <Attributes />
          </View>
        </Card>
        
        <Card style={styles.row} collapsedTitle='BIOGRAPHY'>
          <Text>{bio}</Text>
        </Card>

        <Card style={styles.row}>
          <Abilities abilities={abilities} />
        </Card>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    // flexDirection: 'row',
  },
  row: {
    flexDirection: 'row',
  },
  imgHeroWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingVertical: Layout.paddingSmall,
  },

  hype: {

  },
  lore: {

  },
})