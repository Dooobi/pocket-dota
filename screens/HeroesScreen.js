import React from 'react';
import { View, Text } from 'react-native'

import { SCREEN_LABELS } from '../constants/Constants';
import ButtonHamburger from '../components/ButtonHamburger';

export default class Screen extends React.Component {
  static navigationOptions = ({ navigation }) => ({
    title: SCREEN_LABELS.HEROES,
    headerRight: <ButtonHamburger />
  });

  render() {
    return (
      <View>
        <Text>Heroes content</Text>
      </View>
    );
  }
}

