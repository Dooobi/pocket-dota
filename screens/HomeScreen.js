import React from 'react';
import { Image, Text, TouchableOpacity, View, StyleSheet, } from 'react-native';
import { Thumbnail } from 'native-base';
import { connect } from 'react-redux';

import Colors from '../constants/Colors';
import { HOME_LABELS, SCREEN_LABELS } from '../constants/Constants';


const MenuItem = ({onPress, label, cardImage, profileImage}) => (
  <View style={styles.menuRow}>
    <TouchableOpacity style={styles.imageWrapper} onPress={onPress}>
      <Image style={styles.image} source={cardImage} />
    </TouchableOpacity>

    { !profileImage ? null : 
        <View style={styles.profileImageWrapper} pointerEvents='none'> 
          <Thumbnail large style={styles.profileImage} source={{uri: profileImage}} pointerEvents='none'/>
        </View> 
    }

    <View style={styles.textWrapper} pointerEvents='none'>
      <Text style={styles.text}>{label}</Text>
    </View>
  </View>
);

@connect(state => ({
  profile: state.profile,
}))
export default class Screen extends React.PureComponent {
  static defaultProps = {
    profileName: HOME_LABELS.PROFILE,
  };

  render() {
    const { name, image } = this.props.profile;
    const { navigate } = this.props.navigation;

    return (
      <View style={styles.container}>
        <MenuItem onPress={() => navigate(SCREEN_LABELS.HEROES)} label={HOME_LABELS.HEROES}
          cardImage={require("../assets/images/menu-heroes.png")} />
        <MenuItem onPress={() => navigate(SCREEN_LABELS.ITEMS)} label={HOME_LABELS.ITEMS}
          cardImage={require("../assets/images/menu-items.png")} />
        <MenuItem onPress={() => navigate(SCREEN_LABELS.STATS)} label={HOME_LABELS.STATS}
          cardImage={require("../assets/images/menu-stats.png")} />
        { !image ? null :
          <MenuItem onPress={() => navigate(SCREEN_LABELS.PROFILE)} label={name}
            cardImage={require("../assets/images/menu-profile.png")}
            profileImage={image} /> 
        }
      </View>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: Colors.dota_ui2,
  },
  menuRow: {
    flex: 1,
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: Colors.dota_red_dark,
  },
  imageWrapper: {
    flex: 1,
  },
  image: {
    width: '100%',
    height: '100%',
    resizeMode: 'cover',
  },
  profileImage: {
    marginTop: 5,
  },
  profileImageWrapper: {
    position: 'absolute',
    width:'100%',
    height: '100%',
    
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  textWrapper: {
    right: 20,
    bottom: 20,
    position: 'absolute',
  },
  text: {
    color: 'white',
  },
});