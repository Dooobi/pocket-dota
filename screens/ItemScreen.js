import React from 'react';
import { Container, Text } from '../components/ui';

import ButtonHamburger from '../components/ButtonHamburger';

export default class Screen extends React.Component {
  static navigationOptions = ({ navigation }) => ({
    title: navigation.getParam('itemName'),
    headerRight: <ButtonHamburger />,
  });

  render() {

    return (
      <Container>
          <Text>
            Item description
          </Text>>
      </Container>
    );
  }
}