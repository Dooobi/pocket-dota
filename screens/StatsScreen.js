import React from 'react';
import { Container, Text } from '../components/ui';
import ButtonHamburger from '../components/ButtonHamburger';
import { SCREEN_LABELS } from '../constants/Constants';

export default class Screen extends React.Component {
  static navigationOptions = ({ navigation }) => ({
    title: SCREEN_LABELS.STATS,
    headerRight: <ButtonHamburger />,
  });

  render() {
    return (
      <Container>
        <Text>Stats content</Text>
      </Container>
    );
  }
}
