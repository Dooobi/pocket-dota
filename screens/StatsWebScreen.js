import React from 'react';
import { Container, Text } from '../components/ui';

import ButtonHamburger from '../components/ButtonHamburger';
import { SCREEN_LABELS } from '../constants/Constants';

export default class StatsWebScreen extends React.Component {
  static navigationOptions = ({ navigation }) => ({
    title: SCREEN_LABELS.STATS,
    headerRight: <ButtonHamburger />,
  });

  render() {
    return (
      <Container>
        <Text>Stats webview</Text>
      </Container>
    );
  }
}
