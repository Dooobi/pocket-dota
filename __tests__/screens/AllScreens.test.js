import React from 'react';
import renderer from 'react-test-renderer';
import { store_simple, navigation, getHeroData, getItemData, getPatchName, } from '../__setup__/config';

import AboutScreen from '../../src/screens/AboutScreen';
import HeroesScreen from '../../src/screens/HeroesScreen';
import HeroScreen from '../../src/screens/HeroScreen';
import HomeScreen from '../../src/screens/HomeScreen';
import ItemsScreen from '../../src/screens/ItemsScreen';
import ItemScreen from '../../src/screens/ItemScreen';
import PatchNotesScreen from '../../src/screens/PatchNotesScreen';
import PatchScreen from '../../src/screens/PatchScreen';
import SettingsScreen from '../../src/screens/SettingsScreen';
import SettingsTipsScreen from '../../src/screens/SettingsTipsScreen';
import StatsScreen from '../../src/screens/StatsScreen';
import StatsWebScreen from '../../src/screens/StatsWebScreen';
import TipsScreen from '../../src/screens/TipsScreen';


const screens = [
  AboutScreen, HeroesScreen, HeroScreen,
  HomeScreen, ItemsScreen, ItemScreen,
  PatchNotesScreen, PatchScreen, SettingsScreen,
  SettingsTipsScreen, StatsScreen, StatsWebScreen,
  TipsScreen,
];
export const itRendersScreen = (Screen) => {
  const needsStore = Screen.WrappedComponent;
  const name = needsStore ? Screen.WrappedComponent.name : Screen.name;
  
  let navigationData = {};
  if(Screen == HeroScreen) navigationData = getHeroData();
  if(Screen == ItemScreen) navigationData = getItemData();
  if(Screen == PatchScreen) navigationData = getPatchName();

  it(`${name}`, async () => {
    const tree = renderer.create(
      <Screen 
        store={needsStore ? store_simple : null}
        navigation={navigation(navigationData)}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}


describe("renders screen", async () => {
  screens.forEach(Screen => itRendersScreen(Screen));
});