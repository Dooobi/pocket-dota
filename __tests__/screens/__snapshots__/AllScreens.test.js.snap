// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders screen AboutScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen HeroScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen HeroesScreen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#2F363D",
        "flex": 1,
      },
      Object {
        "padding": undefined,
        "paddingTop": undefined,
      },
      Object {
        "paddingHorizontal": 5,
      },
    ]
  }
/>
`;

exports[`renders screen HomeScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen ItemScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen Itemscreen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#2F363D",
        "flex": 1,
      },
      Object {
        "padding": undefined,
        "paddingTop": undefined,
      },
      Object {
        "paddingHorizontal": 5,
      },
    ]
  }
/>
`;

exports[`renders screen PatchNotesScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen PatchScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen SettingsScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen SettingsTipsScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;

exports[`renders screen StatsScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#2F363D",
          "flex": 1,
        },
        Object {
          "padding": undefined,
          "paddingTop": undefined,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 10,
          "padding": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        enablesReturnKeyAutomatically={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Type in profile name to look up"
        returnKeyType="search"
        selectTextOnFocus={true}
        style={
          Object {
            "backgroundColor": "#d6d6d6",
            "borderRadius": 5,
            "flex": 1,
            "height": 38,
            "marginRight": 10,
            "padding": 10,
          }
        }
        value=""
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1B1E21",
            "borderRadius": 3,
            "height": 38,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#d6d6d6",
              },
              Object {},
              undefined,
            ]
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders screen StatsWebScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#2F363D",
          "flex": 1,
        },
        Object {
          "padding": undefined,
          "paddingTop": undefined,
        },
        undefined,
      ]
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      onSwipe={[Function]}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeDirection="down"
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
        testID={undefined}
      />
      <View
        collapsable={undefined}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onSwipe={[Function]}
        pointerEvents="box-none"
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "left": 0,
            "margin": 37.5,
            "top": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeDirection="down"
        swipeThreshold={100}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#2F363D",
                "flex": 1,
              },
              Object {
                "padding": undefined,
                "paddingTop": undefined,
              },
              Object {
                "backgroundColor": "#1B1E21",
                "borderRadius": 3,
                "flex": 1,
                "padding": 10,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#d6d6d6",
                },
                Object {},
                Object {
                  "color": "#B44335",
                  "fontSize": 15,
                  "marginBottom": 15,
                  "marginTop": 5,
                  "textAlign": "center",
                },
              ]
            }
          >
            The Dota 2 player statistics screen
          </Text>
          <View
            linkText={[Function]}
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#d6d6d6",
                  },
                  Object {},
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 15,
                  },
                ]
              }
            >
              <Text
                URLS={
                  Object {
                    "Dota buff": "https://github.com/dotabuff/d2vpkr",
                    "Elo": "https://elo.io/",
                    "Expo": "https://expo.io/",
                    "Feature Requests": "https://discord.gg/mwP3FRM",
                    "General": "https://discord.gg/WXjby2w",
                    "Issues": "https://discord.gg/BMCrMbw",
                    "NodeJS": "https://nodejs.org/en/",
                    "Open Dota": "https://www.opendota.com",
                    "React Native": "https://facebook.github.io/react-native/",
                    "Valve corporation": "https://www.valvesoftware.com",
                  }
                }
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                hasUrl={true}
                style={
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 15,
                  }
                }
              >
                The Dota 2 player profiles are a collection of statistics gathered and parsed by the 
                <Text
                  URLS={
                    Object {
                      "Dota buff": "https://github.com/dotabuff/d2vpkr",
                      "Elo": "https://elo.io/",
                      "Expo": "https://expo.io/",
                      "Feature Requests": "https://discord.gg/mwP3FRM",
                      "General": "https://discord.gg/WXjby2w",
                      "Issues": "https://discord.gg/BMCrMbw",
                      "NodeJS": "https://nodejs.org/en/",
                      "Open Dota": "https://www.opendota.com",
                      "React Native": "https://facebook.github.io/react-native/",
                      "Valve corporation": "https://www.valvesoftware.com",
                    }
                  }
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  hasUrl={true}
                  onLongPress={undefined}
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "marginBottom": 15,
                      },
                      Object {
                        "color": "#257DAE",
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                >
                  Open Dota
                </Text>
                 platform.
These statistics show an in-depth analysis of the player's activity in the Dota 2 game.

Because the statistics screen is a preview of the Open Dota website, you will need the following buttons in order to navigate the Dota 2 player profiles:
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 15,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#1B1E21",
                  "borderBottomColor": "#2F363D",
                  "borderColor": "#2F363D",
                  "borderTopColor": "#2F363D",
                  "borderWidth": 0,
                  "flex": undefined,
                  "height": 32,
                  "justifyContent": "center",
                  "marginHorizontal": 15,
                  "opacity": 1,
                  "padding": 5,
                  "paddingHorizontal": 10,
                  "width": 37,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#d6d6d6",
                  },
                  Object {},
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              Use the BACK button to navigate back in the current (statistics) screen.
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 15,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#1B1E21",
                  "borderBottomColor": "#2F363D",
                  "borderColor": "#2F363D",
                  "borderTopColor": "#2F363D",
                  "borderWidth": 0,
                  "flex": undefined,
                  "height": 32,
                  "justifyContent": "center",
                  "marginHorizontal": 15,
                  "opacity": 1,
                  "padding": 5,
                  "paddingHorizontal": 10,
                  "width": 37,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#d6d6d6",
                  },
                  Object {},
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              Use the FORWARD button to navigate back in the current (statistics) screen.
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 15,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#1B1E21",
                  "borderBottomColor": "#2F363D",
                  "borderColor": "#2F363D",
                  "borderTopColor": "#2F363D",
                  "borderWidth": 0,
                  "flex": undefined,
                  "height": 32,
                  "justifyContent": "center",
                  "marginHorizontal": 15,
                  "opacity": 1,
                  "padding": 5,
                  "paddingHorizontal": 10,
                  "width": 37,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#d6d6d6",
                  },
                  Object {},
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#d6d6d6",
                    },
                    Object {},
                    undefined,
                  ]
                }
              >
                When the PROFILE button is pressed, it will be 
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#d6d6d6",
                      },
                      Object {},
                      Object {
                        "color": "#DAA520",
                      },
                    ]
                  }
                >
                  highlighted
                </Text>
                , indicating that the current Dota 2 profile is the one that will appear on the app's home screen.
              </Text>
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#d6d6d6",
                },
                Object {},
                undefined,
              ]
            }
          >
            To set another player profile, simply navigate to another Dota 2 profile (from the Stats screen) and press the PROFILE button.
          </Text>
          <View
            style={
              Object {
                "alignSelf": "center",
                "bottom": 5,
                "position": "absolute",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#1B1E21",
                  "borderBottomColor": "#2F363D",
                  "borderColor": "#2F363D",
                  "borderTopColor": "#2F363D",
                  "borderWidth": 1,
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#d6d6d6",
                    },
                    Array [
                      undefined,
                      Object {
                        "color": "#d6d6d6",
                      },
                    ],
                    undefined,
                  ]
                }
              >
                DONE
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTWebView
        allowsInlineMediaPlayback={undefined}
        automaticallyAdjustContentInsets={undefined}
        bounces={undefined}
        contentInset={undefined}
        dataDetectorTypes={undefined}
        decelerationRate={undefined}
        injectedJavaScript="
  var css = \\"body { background-color:#2F363D!important; } #root { background-color:#2F363D!important; background-image:none; } ul { background-color:#1B1E21!important; } .gauge-container { background-color:#1B1E21!important; } thead { background-color:#1b1e21b5!important; }\\";
  var head = document.getElementsByTagName(\\"head\\")[0];
  var style = document.createElement(\\"style\\");
  style.appendChild(document.createTextNode(css));
  head.appendChild(style);

  var meta = document.createElement('meta'); 
  meta.setAttribute('name', 'viewport'); 
  meta.setAttribute('content', 'width=device-width, user-scalable=0');
  head.appendChild(meta);

  var main = document.getElementById(\\"root\\").childNodes[0];

  main.childNodes[main.childNodes.length-2].remove();
  main.childNodes[main.childNodes.length-1].remove();
  main.childNodes[1].remove();
  main.childNodes[0].remove();
  
  function playerInfo() {
    var playerInfoDiv = document.getElementsByClassName('playerInfo')[0];
    if(playerInfoDiv) {
      playerInfoDiv.childNodes[2].remove();
      playerInfoDiv.childNodes[0].childNodes[1].remove();
      clearInterval(playerInfoInterval);
    }
  }
  var playerInfoInterval = setInterval(playerInfo, 200);

"
        mediaPlaybackRequiresUserAction={undefined}
        messagingEnabled={false}
        onLoadingError={[Function]}
        onLoadingFinish={[Function]}
        onLoadingStart={[Function]}
        onMessage={[Function]}
        onShouldStartLoadWithRequest={undefined}
        scalesPageToFit={true}
        scrollEnabled={undefined}
        source={
          Object {
            "uri": "https://www.opendota.com/players/undefined/overview",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            Object {
              "backgroundColor": "#2F363D",
            },
            Object {
              "flex": 0,
              "height": 0,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#DAA520"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#1B1E21",
          "borderTopColor": "#777777",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1B1E21",
            "borderBottomColor": "#2F363D",
            "borderTopColor": "#2F363D",
            "borderWidth": 0,
            "height": 32,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
            "width": 172.5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1B1E21",
            "borderBottomColor": "#2F363D",
            "borderTopColor": "#2F363D",
            "borderWidth": 0,
            "height": 32,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
            "width": 172.5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1B1E21",
            "borderBottomColor": "#2F363D",
            "borderTopColor": "#2F363D",
            "borderWidth": 0,
            "height": 32,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
            "width": 172.5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1B1E21",
            "borderBottomColor": "#2F363D",
            "borderTopColor": "#2F363D",
            "borderWidth": 0,
            "height": 32,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 5,
            "width": 172.5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders screen TipsScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2F363D",
      "flex": 1,
    }
  }
/>
`;
